test_cases/stdlib/builtins/check_exception_group.py

```diff
@@ -0,0 +1,327 @@
+from __future__ import annotations
+
+import sys
+from typing import TypeVar
+from typing_extensions import assert_type
+
+if sys.version_info >= (3, 11):
+    # BaseExceptionGroup
+    # ==================
+
+    # `BaseExceptionGroup` can work with `BaseException`:
+    beg = BaseExceptionGroup("x", [SystemExit(), SystemExit()])
+    assert_type(beg, BaseExceptionGroup[SystemExit])
+    assert_type(beg.exceptions, tuple[SystemExit | BaseExceptionGroup[SystemExit], ...])
+
+    # Covariance works:
+    _beg1: BaseExceptionGroup[BaseException] = beg
+
+    # `BaseExceptionGroup` can work with `Exception`:
+    beg2 = BaseExceptionGroup("x", [ValueError()])
+    # FIXME: this is not right, runtime returns `ExceptionGroup` instance instead,
```

Comment:
Should be doable by overriding `__new__` (python/mypy#7188), did you try that?

Reply:
We have three cases:
- `BaseExceptionGroup.__new__` can return `BaseExceptionGroup` for `_BaseExceptionT_co`
- `BaseExceptionGroup.__new__` can return `ExceptionGroup` for `_ExceptionT_co`
- `BaseExceptionGroup.__new__` can return custom subclass

Either the first two work or the third one.